presto-spi/src/main/java/com/facebook/presto/spi/page/PagesSerdeUtil.java

```diff
@@ -107,8 +113,8 @@ public static long computeSerializedPageChecksum(Slice pageData, byte markers, i
         }
         crc32.update(pageData.byteArray(), pageData.byteArrayOffset(), pageData.length());
         crc32.update(markers);
-        crc32.update(positionCount);
-        crc32.update(uncompressedSize);
+        crc32.update(toBytes(positionCount));
```

Comment:
`Integer.reverseBytes` to avoid extra allocations (static import `reverseBytes`)

Reply:
Discussed offline with @mbasmanova . The CRC interface take's an integer, but only adds the last byte of an integer to the checksum. I'm thinking if we can avoid an extra allocation. Maybe we an feed the integer byte-by-byte to the checksum?

Reply:
@arhimondr Address this issue by calculating crc iteratively.